### PR TITLE
WIP: Migrate H3 website to ocular-gatsby

### PR DESCRIPTION
**Pull request by isaacbrodsky**
The version of Ocular currently used in H3 is no longer supported and has been replaced by an Ocular implementation based on [Gatsby](https://www.gatsbyjs.org/). This requires a different configuration.

Some benefits of this change:
* Upgrades us to the latest version supported by the Uber team
* No longer uses anchor links, which I expect to help with navigability

This PR has a few issues with it still. Local development of the website is slightly broken because image path prefixing does not seem to be correctly setup. The image paths are hardcoded to use the prefix. I'm also not entirely certain the `a href` paths are being resolved correctly. It looks like the links inside documents are missing the prefix as well.

### CODE REVIEW

File: website/gatsby-config.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+          desc: 'Along with twelve pentagons, the entire world is addressable in H3, down to square meter resolution.',
+          img: 'images/icon-chart.svg'
+        }
+      ],
+
+      // your table of contents goes here
+      DOCS,
+
+      EXAMPLES: [
+        //{
```
Comment (ibgreen) [inline website/gatsby-config.js]: Cut or comment?

**Comment by isaacbrodsky** [comment]
>     * Does removing `src` from `SOURCE` config not work? regarding the `.gitkeep`?

Unfortunately that does not build:

```
 ERROR 


The path passed to gatsby-source-filesystem does not exist on your file system:
<...>/h3/website/src/
Please pick a path to an existing directory.
See docs here - https://www.gatsbyjs.org/packages/gatsby-source-filesystem/
```

**Comment by isaacbrodsky** [comment]
This seems to be working well now, with the exception of the Katex plugin not working; it creates duplicate text - I think this is because the CSS is not included correctly. I am no longer trying to get path prefixing to work and assume we'll host it on h3geo.org

File: website/gatsby-config.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      ROOT_FOLDER: `${__dirname}/../`,
+      DIR_NAME: `${__dirname}`,
+      SOURCE: [`${__dirname}/static`, `${__dirname}/src`],
+
+      PROJECT_TYPE: 'github',
+      PROJECT_NAME: 'H3',
+      PROJECT_ORG: 'uber',
+      PROJECT_URL: 'https://github.com/uber/h3',
+      PROJECT_DESC: 'Hexagonal hierarchical geospatial indexing system.',
+      PATH_PREFIX: '/',
```
Comment (Pessimistress) [inline website/gatsby-config.js]: If the website is still at `https://uber.github.io/h3/` then `PATH_PREFIX` needs to be `/h3`.

I do not believe this is working in the current prod version of gatsby-theme-ocular. A bunch of fixes (as well as breaking changes) are made in 1.2.0-beta.

Comment (isaacbrodsky) [reply website/gatsby-config.js]: The H3 website is moving to https://h3geo.org so it shouldn't need a prefix anymore. I had difficulty getting the prefixing to work consistently in ocular-gatsby, e.g. I couldn't get it to work for all of images and links.

File: yarn.lock [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (isaacbrodsky) [inline yarn.lock]: I don't think this file should have been included in the PR - looks like a spurious lockfile?

**Pull request merged**